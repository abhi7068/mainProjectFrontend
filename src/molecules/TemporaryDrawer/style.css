/* width */
::-webkit-scrollbar {
  width: 10px;
}

/* Track */
::-webkit-scrollbar-track {
  box-shadow: inset 0 0 5px #969397;
  border-radius: 10px;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: #969397;
  border-radius: 10px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #5f5e5f;
}
