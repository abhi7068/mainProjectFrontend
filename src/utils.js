import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
