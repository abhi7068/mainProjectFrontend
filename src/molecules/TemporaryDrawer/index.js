import React, { useEffect } from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { getCategories } from '../../services/categoryServices';
import { sortByPrice } from '../../store/actions/productAction';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import kebab, { firstCapital } from '../../services/kebab';
import MenuIcon from '@material-ui/icons/Menu';
import DrawerCompoent from '../DrawerCompoent';
import './style.css';
const drawerWidth = 300;

const useStyles = makeStyles((theme) => ({
  clickable: {
    cursor: 'pointer',
    '&:hover, &:focus': {
      color: 'red',
    },
    '&:active': {
      color: 'red',
    },
  },
  list: {
    width: 250,
  },
  fullList: {
    width: 'auto',
  },
  root: {
    display: 'flex',
  },
  nested: {
    paddingLeft: theme.spacing(4),
  },
  listItemText: {
    fontSize: '0.9em',
    color: 'white', // Insert your required size
  },
  size: {
    fontSize: '18px',
  },
  drawerPaper: {
    width: drawerWidth,
    // backgroundImage: 'url(' + stars + ')',
  },
  paper: {
    flexDirection: 'column',
  },
}));
const TemporaryDrawer = (props) => {
  const classes = useStyles();
  const [state, setState] = React.useState({
    left: false,
  });

  useEffect(() => {
    props.getCategories();
  }, []);
  const handleSort = () => {
    props.sortByPrice();
  };

  const toggleDrawer = (anchor, open) => (event) => {
    if (
      event.type === 'keydown' &&
      (event.key === 'Tab' || event.key === 'Shift')
    ) {
      return;
    }

    setState({ ...state, [anchor]: open });
  };

  // Sorting
  const list = (anchor) => (
    <div>
      <ListItem style={{ color: '#969397' }}>
        <ListItemIcon style={{ color: 'Black' }}></ListItemIcon>
        <ListItemText style={{ color: '#969397' }} primary="Categories" />
      </ListItem>
      <List component="div">
        {props.categories.data.map((category, index) => {
          return (
            <div key={index}>
              <ListItem className={clsx(classes.nested)}>
                <ListItemText style={{ marginLeft: '  36px' }}>
                  <div>
                    {' '}
                    <NavLink
                      color="primary"
                      noWrap
                      activeClassName="is-active"
                      activeStyle={{ color: 'red' }}
                      variant="body2"
                      className={classes.toolbarLink}
                      to={`/categories/${kebab(category.title)}/${
                        category._id
                      }`}
                      label="active"
                    >
                      {firstCapital(category.title)}
                    </NavLink>
                  </div>
                </ListItemText>
              </ListItem>
            </div>
          );
        })}
      </List>
      <ListItem>
        <ListItemIcon style={{ color: 'Black' }}></ListItemIcon>
        <ListItemText style={{ color: '#969397' }} primary="Sorting" />
      </ListItem>
      <List component="div">
        <ListItem button onClick={handleSort}>
          <ListItemIcon style={{ color: 'black' }}></ListItemIcon>
          <ListItemText style={{ color: 'black' }} primary="Sort By Price" />
        </ListItem>
      </List>
      <ListItem>
        <ListItemIcon style={{ color: 'Black' }}></ListItemIcon>
        <ListItemText style={{ color: '#969397' }} primary="Fliters" />
      </ListItem>
      <DrawerCompoent />
    </div>
  );
  return (
    <div>
      {['left'].map((anchor) => (
        <React.Fragment key={anchor}>
          <Button data-testid="test-input" onClick={toggleDrawer(anchor, true)}>
            <MenuIcon style={{ color: 'white  ' }} />
          </Button>
          <Drawer
            anchor={anchor}
            open={state[anchor]}
            onClose={toggleDrawer(anchor, false)}
            classes={{
              paper: clsx(classes.drawerPaper),
            }}
          >
            {list(anchor)}
          </Drawer>
        </React.Fragment>
      ))}
    </div>
  );
};
const mapStateToProps = (state) => {
  return {
    state: state,
    categories: state.categories,
  };
};
TemporaryDrawer.propTypes = {
  getCategories: PropTypes.func,
  sortByPrice: PropTypes.func,
  categories: PropTypes.array,
  getResetSorting: PropTypes.func,
};

export default connect(mapStateToProps, {
  getCategories,
  sortByPrice,
})(TemporaryDrawer);
