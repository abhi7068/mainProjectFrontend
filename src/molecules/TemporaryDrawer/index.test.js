import React from 'react';
import { render, cleanup, fireEvent } from '../../utils';
import TemporaryDrawer from '../TemporaryDrawer';

afterEach(cleanup);
it('renders', () => {
  const { asFragment } = render(<TemporaryDrawer />);
  expect(asFragment(<TemporaryDrawer />)).toMatchSnapshot();
});
it('renders', () => {
  const { getByTestId } = render(<TemporaryDrawer />);
  expect(fireEvent.click(getByTestId('test-input'))).toBe(true);
});
