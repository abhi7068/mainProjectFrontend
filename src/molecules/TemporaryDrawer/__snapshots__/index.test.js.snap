// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      data-testid="test-input"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style="color: white;"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
